doc/sr_fortran_walkthrough.rst

```diff
@@ -21,6 +21,22 @@ SmartRedis ``DataSet`` API is also provided.
       Update the ``Client`` constructor ``cluster`` flag to `.false.`
       to connect to a single shard (single compute host) database.
 
+Error handling
+==============
+
+The core of the SmartRedis library is written in C++ which utilizes the
+exception handling features of the language to catch errors. This same
+functionality does not exist in Fortran, so instead most SmartRedis
+methods are functions that return error codes that can be checked. This
+also has the added benefit that Fortran programs can incorporate
+SmartRedis calls within their own error handling methods. By convention,
+a ``0`` return code represents a successful completion, whereas non-zero
```

Comment:
Please don't refer to enum members by their values -- it limits our ability to change them in the future. Document instead that SRNoError is the "all is OK" value.